src/bdc/steps/google_places.py

```diff
@@ -34,50 +69,212 @@ def verify(self) -> bool:
+        results_list.append(self.calculate_confidence(results_list, lead_row))
+
+        return pd.Series(results_list)
+
+    def get_data_from_detailed_google_api(self, lead_row):
+        error_return_value = pd.Series([None] * len(self.detailed_df_fields))
+
+        place_id = lead_row["google_places_place_id"]
+
+        if place_id is None or pd.isna(place_id):
             return error_return_value
 
+        # Call for the detailed API using specified fields
         try:
-            response = self.gmaps.find_place(domain, "textquery", fields=self.fields)
+            # Fetch place details including reviews
+            response = self.gmaps.place(place_id, fields=self.detailed_api_fields)
+
+            # Check response status
+            if response.get("status") != HTTPStatus.OK.name:
+                self.log(f"Failed to fetch data. Status code: {response.get('status')}")
+                return error_return_value
+
+        except RequestException as e:
+            self.log(f"Error: {str(e)}")
+
+        except (ApiError, HTTPError, Timeout, TransportError) as e:
+            error_message = (
+                str(e.message)
+                if hasattr(e, "message") and e.message is not None
+                else str(e)
+            )
+            self.log(f"Error: {error_message}")
+
+        results_list = [
+            response["result"][field] if field in response["result"] else None
+            for field in self.detailed_api_fields_output
+        ]
+
+        return pd.Series(results_list)
+
+    def get_first_place_candidate(self, query, input_type) -> (dict, int):
+        if query is None:
+            return None, 0
+        try:
+            response = self.gmaps.find_place(query, input_type, fields=self.api_fields)
+
             # Retrieve response
             # response = requests.get(self.URL + domain + "&key=" + GOOGLE_PLACES_API_KEY)
         except RequestException as e:
             self.log(f"Error: {str(e)}")
-            return error_return_value
+            return None, 0
+        except (ApiError, HTTPError, Timeout, TransportError) as e:
+            self.log(f"Error: {str(e.message) if e.message is not None else str(e)}")
+            return None, 0
 
         if not response["status"] == HTTPStatus.OK.name:
             self.log(f"Failed to fetch data. Status code: {response['status']}")
-            return error_return_value
+            return None, 0
 
         if "candidates" not in response or len(response["candidates"]) == 0:
-            return error_return_value
+            return None, 0
 
-        # Only look at the top result TODO: Check if we can cross check available values to rate results
         top_result = response["candidates"][0]
 
-        results_list = [
-            top_result[field] if field in top_result else None for field in self.fields
-        ]
+        # database connection TODO: replace this connection to appropriate file
```

Comment:
this should probably not happen here, but rather in pipeline.py or a seperate step